packages/react-image/src/components/Image/useImage.ts

```diff
@@ -23,6 +24,26 @@ export const useImage = (
   options: ComposePreparedOptions,
 ): ImageState => {
   const imageRef = React.useRef<HTMLElement>(null);
+  const ariaProps = useImageAria(props);
+
+  const handleImageLoad: React.DOMAttributes<HTMLImageElement>['onLoad'] = e => {
+    if (props.onLoad) {
+      props.onLoad(e);
+    }
+
+    if (imageRef.current) {
+      const rect = imageRef.current.getBoundingClientRect();
+      let actualImageSize: 'small' | 'medium' | 'large' = 'small';
+
+      if (rect.width > 80) {
+        actualImageSize = 'large';
+      } else if (rect.width > 32) {
+        actualImageSize = 'medium';
+      }
+
+      imageRef.current.setAttribute('data-image-size', actualImageSize);
```

Comment:
_This is the most controversial part 🤔_  The design spec requires to have different border sizes (see `Image.scss`) for different image sizes.

- Is it a valid requirement at all?
- Should be this a part of `BaseImage` component or only of `Image`?
- Better solutions? 😺

Reply:
/cc @khmakoto @dzearing

Reply:
This seems like a bad requirement unless it were handled universally on every component and likely should be handled in the parent styling. I am not understanding the use case. It seems like it would depend on the use case to define.

I'd suggest we ask the designer for use cases they had in mind and understand if this is more a higher level concern or really belongs in the `Image`. I thought of the `Image` more as a utility.

Reply:
I have an approval from design, we will use `medium` as a default. We will not use this logic.